volatility3/framework/plugins/windows/ldrmodules.py

```diff
@@ -0,0 +1,97 @@
+            requirements.VersionRequirement(name = 'vadinfo', component = vadinfo.VadInfo, version = (2, 0, 0)),
+            requirements.ListRequirement(name = 'pid',
+                                         element_type = int,
+                                         description = "Process IDs to include (all other processes are excluded)",
+                                         optional = True),
+           ]
+
+    def _generator(self, procs):
+
+        pe_table_name = intermed.IntermediateSymbolTable.create(self.context,
+                                                                self.config_path,
+                                                                "windows",
+                                                                "pe",
+                                                                class_types = pe.class_types)
+
+        def filter_function(x: interfaces.objects.ObjectInterface) -> bool:
+            try:
+                return not (x.get_private_memory() == 0 and x.ControlArea)
+            except AttributeError:
+                return False
+
+        filter_func = filter_function   
+
+        for proc in procs:
+            proc_layer_name = proc.add_process_layer()
+
+            load_order_mod = dict((mod.DllBase, mod)
+                                for mod in proc.load_order_modules())
+            init_order_mod = dict((mod.DllBase, mod)
+                                for mod in proc.init_order_modules())
+            mem_order_mod = dict((mod.DllBase, mod)
+                                for mod in proc.mem_order_modules())
+
+            mapped_files = {}
+            for vad in vadinfo.VadInfo.list_vads(proc, filter_func = filter_func):
+                dos_header = self.context.object(pe_table_name + constants.BANG + "_IMAGE_DOS_HEADER",
+                                   offset = vad.get_start(),
+                                   layer_name = proc_layer_name)
+                try:
+                    if dos_header.e_magic != 0x5A4D:
+                        continue
+                except exceptions.PagedInvalidAddressException:
+                    continue
+
+                mapped_files[int(vad.get_start())] = str(vad.get_file_name() or '')
+
+            for base in mapped_files.keys():
+                # Does the base address exist in the PEB DLL lists?
+                load_mod = load_order_mod.get(base, None)
+                init_mod = init_order_mod.get(base, None)
+                mem_mod = mem_order_mod.get(base, None)
+
+                yield (0, [int(proc.UniqueProcessId),
+                            str(proc.ImageFileName.cast("string",
+                                                max_length = proc.ImageFileName.vol.count,
+                                                errors = 'replace')),
+                            format_hints.Hex(base),
+                            str(load_mod != None),
```

Comment:
If you don't want to return the actual value of these things (and use a `BaseAbsentValue` for when it's not there), then it would be better to make this a bool type, so that another UI can directly make use of the value if it wants, rather than having to parse a string.